### PR TITLE
feat: add route as a separate screen template parameter #178

### PR DESCRIPTION
**Pull request by ilya-trifonov**
affects: @amplicode/codegen

### CODE REVIEW

**Comment by vyacheslav-pushkin** [comment]
@ilya-trifonov please resolve conflicts

File: packages/codegen/src/generators/react-typescript/entity-management/answers.ts [hunk 1]
```diff
@@ -25,6 +26,16 @@ export const commonEntityManagementQuestions =  [
...
+  {
+    caption: 'List component route',
+    code: 'route',
+    propertyType: StudioTemplatePropertyType.ROUTE,
+    required: true,
+    step: {
+      name: "Entity List",
+      order: "1"
+    }
+  },
```
Comment (vyacheslav-pushkin) [inline packages/codegen/src/generators/react-typescript/entity-management/answers.ts:38]: It should have `relatedProperty` pointing to component name.

File: packages/codegen/src/building-blocks/stages/answers/amplicode/ScreenAnswers.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export interface ScreenAnswers {
   componentName: string,
+  route?: string,
```
Comment (vyacheslav-pushkin) [inline packages/codegen/src/building-blocks/stages/answers/amplicode/ScreenAnswers.ts:3]: It should be mandatory. Studio will provide the default.

File: packages/codegen/src/building-blocks/stages/answers/pieces/defaultAnswers.ts [hunk 3]
```diff
@@ -56,3 +56,20 @@ export const createQueryQuestion = (options?: QueryQuestionOptions) => ({
...
+    required?: boolean;
+    relatedProperty?: string;
+}
+export const createRoute = (options: routeOptions) => ({
+    code: options?.code || 'route',
+    caption: options?.caption || 'Component route',
+    propertyType: StudioTemplatePropertyType.ROUTE,
+    relatedProperty: options?.relatedProperty || "componentName",
+    required: options?.required || true,
+});
```
Comment (vyacheslav-pushkin) [inline packages/codegen/src/building-blocks/stages/answers/pieces/defaultAnswers.ts:75]: Why do we need all this?

File: packages/codegen/src/building-blocks/stages/writing/pieces/mvp/addMvpAppMenu.ts [hunk 4]
```diff
@@ -31,8 +31,10 @@ export function addMvpAppMenu({
     return;
   }
   const appMenuContents = gen.fs.read(appMenuPath);
-  const appMenuTransformed = transformAddMenuItem(appMenuContents, route);
-  gen.fs.write(appMenuPath, appMenuTransformed);
+  if (route != null) {
```
Comment (vyacheslav-pushkin) [inline packages/codegen/src/building-blocks/stages/writing/pieces/mvp/addMvpAppMenu.ts:34]: `route` should be mandatory

File: packages/codegen/src/generators/react-typescript/entity-list/answers.ts [hunk 5]
```diff
@@ -19,6 +20,12 @@ export const entityListQuestions = [
     defaultValue: 'List',
     required: true
   },
+  {
+    caption: 'Component route',
+    code: 'route',
+    propertyType: StudioTemplatePropertyType.ROUTE,
+    required: true
+  },
```
Comment (vyacheslav-pushkin) [inline packages/codegen/src/generators/react-typescript/entity-list/answers.ts:28]: `relatedProperty` should point to component name

File: scripts/generate-owner-management.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
 const { runCmdSync, esc, btoa, amplicodegen } = require("./common");
+// const {toKebabCase} = require('./../packages/codegen/src/building-blocks/util')
```
Comment (vyacheslav-pushkin) [inline scripts/generate-owner-management.js:2]: Please remove this line

**Review by vyacheslav-pushkin — CHANGES_REQUESTED** [review]

File: packages/codegen/src/building-blocks/stages/answers/amplicode/ScreenAnswers.ts (hunk 2 above)
Comment (ilya-trifonov) [reply packages/codegen/src/building-blocks/stages/answers/amplicode/ScreenAnswers.ts]: route is needed in the list components only, that's why I did it..

**Review by vyacheslav-pushkin — APPROVED** [review]

**Pull request merged**